File: Examples/cosmology/bao.py
## =============================================================
## Example code: how to estimate the sound horizon at drag epoch
## =============================================================

# to ensure compatibility in Python versions 2.x and 3.x
from __future__ import print_function

# import the CosmoBolognaLib module
import CosmoBolognaLib as cbl

# set the cosmological model, using default parameters
cosmology = cbl.LCDM("Planck18")

# create the object used for the computation of the sound horizon at drag epoch
bao = cbl.BAO(cosmology)

print("The sound horizon at drag epoch is:", bao.rs("EisensteinHu"))

File: Examples/cosmology/bao.cpp
// =============================================================
// Example code: how to estimate the sound horizon at drag epoch 
// =============================================================

#include "LCDM.h"
#include "BAO.h"

using namespace std;

int main () {
    
  try {
    
    // --------------------------------------------------------
    // ---------------- set the cosmological model ------------
    // --------------------------------------------------------

    auto cosmology = make_shared<cbl::cosmology::LCDM>("Planck18");
    
    
    // ----------------------------------------------------------------------
    // ---------------- estimate the sound horizon at drag epoch ------------
    // ----------------------------------------------------------------------

    cbl::cosmology::BAO bao(cosmology);

    const double rs = bao.rs("EisensteinHu");

    cout << "The sound horizon at drag epoch is: " << rs << endl;
    
  }

  catch(cbl::glob::Exception &exc) { cerr << exc.what() << endl; exit(1); }

  return 0;
}
